service/pom.xml

```diff
@@ -179,6 +179,37 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.pac4j</groupId>
+      <artifactId>pac4j-saml-opensamlv3</artifactId>
+      <version>4.0.3</version>
```

Comment:
should define the scope for this dependency. Might pull in transitive dependencies that might be undesirable.

Reply:
I found on dependency which can potentially cause problems (commons-collections). I excluded it here. However, finding all the dependencies which are unnecessary may be a bit more complex. I will create a separate JIRA for that if that is okay with you.

Reply:
Created HIVE-24619 to track this.